### PR TITLE
docs: update the doc with HA info

### PR DESCRIPTION
**Pull request by anupriya0703**
Signed-off-by: anupriya0703 <anupriya.gupta@mayadata.io>

### CODE REVIEW

File: reference/HA.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## High Availability 
+
+Mayastor 2.0 enhances High Availability (HA) of the volume target with the nexus switch-over feature. In the event of the target failure, the switch-over feature quickly detects the failure and spawns a new nexus to ensure I/O continuity.
+The HA feature consists of two components: the HA node agent (which runs in each csi- node) and the cluster agent (which runs alongside the agent-core). The HA node agent looks for io-path failures from applications to their corresponding targets. If any such broken path is encountered, the HA node agent informs the cluster agent. The cluster-agent then creates a new target on a different (live) node. Once the target is created, the `csi-node` establishes a new path between the application and its corresponding target. The HA feature restores the broken path within seconds, ensuring negligible downtime. 
+
+{% hint style="warning" %}
+The volume's replica count must be higher than 1 for a new target to be established as part of switch-over.
```
Comment (avishnu) [inline reference/HA.md:7]: Is this true? @tiagolobocastro @Abhinandan-Purkait 
@anupriya0703 we can merge the current PR and take the action (if needed) on this comment in another PR.

**Review by avishnu — APPROVED** [review]
lgtm

File: reference/HA.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## High Availability 
+
+Mayastor 2.0 enhances High Availability (HA) of the volume target with the nexus switch-over feature. In the event of the target failure, the switch-over feature quickly detects the failure and spawns a new nexus to ensure I/O continuity.
+The HA feature consists of two components: the HA node agent (which runs in each csi- node) and the cluster agent (which runs alongside the agent-core). The HA node agent looks for io-path failures from applications to their corresponding targets. If any such broken path is encountered, the HA node agent informs the cluster agent. The cluster-agent then creates a new target on a different (live) node. Once the target is created, the `csi-node` establishes a new path between the application and its corresponding target. The HA feature restores the broken path within seconds, ensuring negligible downtime.
```
Comment (tiagolobocastro) [inline reference/HA.md]: I think `csi-node` in ("the `csi-node` establishes a new path betwee..") can be replaced with `node-agent` for all intents and purposes.

File: reference/HA.md (hunk 1 above)
Comment (tiagolobocastro) [reply reference/HA.md:7]: hmm not necessarily no, that's only the case if the replica is co-located with the target AND the target is down/not responding.

File: reference/HA.md (hunk 2 above)
Comment (anupriya0703) [reply reference/HA.md]: Updated the text.

File: reference/HA.md (hunk 1 above)
Comment (anupriya0703) [reply reference/HA.md:7]: Merging the current PR. If needed, will be addressing this issue in next PR.

**Pull request merged**